src/parsers/forecast_parser.jl

```diff
@@ -293,3 +293,42 @@ function parse_json(filename,device_names)
     return Devices
 end
 =#
+
+"""
+    remove_all_forecasts(sys::System, T::Type{T})::Vector{T} where T <: Forecast
+
+Remove all forecast objects of concrete type T from a System
+"""
+function remove_all_forecasts!(sys::System, ForecastType::Type{T})where T <: Forecast
+
+    ts = get_forecast_initial_times(sys)
+
+    for t in ts
+        f = get_forecasts(ForecastType, sys, t)
+        foreach(x -> remove_forecast!(sys,x),f)
+    end
+
+end
+
+"""
+    split_forecasts!(sys::System, 
+                    forecast::FlattenIteratorWrapper{T}, 
```

Comment:
Enforcing FlattenIteratorWrapper may be too rigid. I could imagine a user calling this with a vector.  I suggest leaving this untyped and put a comment saying that it needs to be an iterable.  I think Julia will eventually add a way to specify an interface type for this case.

Also, forecast should probably be plural.